ts-client/src/dlmm/index.ts

```diff
@@ -3388,14 +3395,56 @@ export class DLMM {
       this.lbPair.binStep
     );
 
+    if (maxExtraBinArrays > 0 && maxExtraBinArrays <= MAX_EXTRA_BIN_ARRAYS) {
+      const extraBinArrays: Array<PublicKey> = new Array<PublicKey>();
+
+      while (extraBinArrays.length < maxExtraBinArrays) {
+        let binArrayAccountToSwap = findNextBinArrayWithLiquidity(
+          swapForY,
+          activeId,
+          this.lbPair,
+          this.binArrayBitmapExtension?.account ?? null,
+          binArrays
+        );
+
+        if (binArrayAccountToSwap == null) {
+          break;
+        }
+
+        const binArrayAccountToSwapExisted = binArraysForSwap.has(binArrayAccountToSwap.publicKey);
+
+        if (binArrayAccountToSwapExisted) {
+          if (swapForY) {
+            activeId = activeId.sub(new BN(1));
+          } else {
+            activeId = activeId.add(new BN(1));
+          }
+        } else {
+          extraBinArrays.push(binArrayAccountToSwap.publicKey);
+          if (swapForY) {
+            activeId = activeId.sub(new BN(binArrayAccountToSwap.account.bins.length));
+          } else {
```

Comment:
activeId = max_bin_id_in(binArrayAccountToSwap) + 1

Reply:
updated to upper bin id in binArray